control-plane/agents/core/src/volume/specs.rs

```diff
@@ -587,6 +607,19 @@ impl ResourceSpecsLocked {
         Ok(volume)
     }
 
+    // Delete the NexusInfo key from the persistent store.
+    // If deletion fails we just log it and continue.
+    async fn delete_nexus_info(key: &NexusInfoKey, registry: &Registry) {
+        match registry.delete_kv(&key.key()).await {
+            Ok(_) => {
+                tracing::info!(nexus_info_key=%key,"Deleted NexusInfo entry from persistent store")
+            }
+            Err(e) => {
+                tracing::error!(nexus_info_key=%key, error=%e, "Failed to delete NexusInfo entry from persistent store")
```

Comment:
Ah this is why you've implemented Display for NexusInfoKey?
That doesn't work nicely with tracing, it's probably better to add the individual fields by hand here; or uses the nexus and just emit a nexus.error which will contain the volume and nexus uuid.

Reply:
:+1: Added them by hand.